tflite2onnx/op/reshape.py

```diff
@@ -72,17 +72,143 @@ def parse(self):
 
         self.setParsed()
 
+    def preserveInputSpatialSemantic(self):
+        # If the input (or output) tensor of Reshape
+        # holds a special data layout semantic,
+        # we need to insert Transpose before (or after) the Reshape operator
+        # https://github.com/jackwish/tflite2onnx/issues/28
+        # An example for inserting `Transpose` before `Reshape`
+        #    ------
+        #   | Conv |
+        #    ------
+        #      |
+        #      |  to_transpose (Original input of `Reshape`)
+        #      |  (e.g. NCHW)
+        #   -------
+        #  | Trans |  e.g. perm: (0, 2, 3, 1)
+        #   -------
+        #      |
+        #      |  transposed (New created tensor)
+        #      |  (e.g. NHWC)
+        #   --------
+        #  | Reshape |
+        #   --------
+        #
+
+        assert(self.status.parsed)
+        to_transpose = self.inputs[0]
+
+        transposed_name = 'TFLITE2ONNX_Transposed_%s' % to_transpose.name
+        transposed = self.TFactory.getWithRef(to_transpose, transposed_name, True)
+
+        # Copy the layout of original input of `Reshape`
+        # e.g. current: NCHW, (source: NHWC, target: NCHW)
+        layout = copy.deepcopy(to_transpose.layout)
+        # Exchange the source and target layout for layout preservation
```

Comment:
Hmmm, maybe we can construct a `Layout` object directly with the parameters from `to_transpose.layout`? And, `Layout.current` doesn't take significant currently if I remember correctly. So mostly we can ignore it.